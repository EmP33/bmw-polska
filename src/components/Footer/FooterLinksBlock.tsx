import React from "react";

interface Props {
  title: string;
  elements: string[];
}

const FooterLinksBlock: React.FC<Props> = ({ title, elements }) => {
  return (
    <div className="footer-links__block">
      <span>{title}</span>
      <ul>
        {elements.map((element) => (
          <li>
            <a href="#">{element}</a>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default FooterLinksBlock;
