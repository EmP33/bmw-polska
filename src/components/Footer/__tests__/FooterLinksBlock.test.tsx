import FooterLinksBlock from "../FooterLinksBlock";
import ReactShallowRenderer from "react-test-renderer/shallow";

describe("<Footer />", () => {
  it("should render Footer properly", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
      <FooterLinksBlock
        title="Informacje prawne"
        elements={[
          "Mapa serwisu",
          "Nota prawna / Prywatność",
          "Polityka cookie",
          "Warunki korzystania z MyBMW",
        ]}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
