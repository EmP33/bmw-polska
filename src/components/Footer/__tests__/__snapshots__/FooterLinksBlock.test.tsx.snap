// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render Footer properly 1`] = `
<div
  className="footer-links__block"
>
  <span>
    Informacje prawne
  </span>
  <ul>
    <li>
      <a
        href="#"
      >
        Mapa serwisu
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Nota prawna / Prywatność
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Polityka cookie
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Warunki korzystania z MyBMW
      </a>
    </li>
  </ul>
</div>
`;
